packages/docs/docs/tutorial/part4-docsqa.md

```diff
@@ -108,3 +109,62 @@ The `<GetChunk>` component is used to format each document chunk as it is presen
 to the LLM. In this case, we just return the raw text of the chunk, but if we wanted
 to include some additional metadata, or transformed the chunks in some way before
 processing them, we could have done so.
+
+## Using a Pinecone database
+
+Until now, we have been using an in-memory Corpus, which is good for a demo, but in practice you might want to use a vector database like [Pinecone](https://www.pinecone.io/) or [Chroma](https://www.trychroma.com/) instead.
+
+To do so, you can use [LangchainCorpus](../api/classes/batteries_docs.LangChainCorpus) to integrate with any [VectorStore from LangChain.js](https://js.langchain.com/docs/modules/indexes/vector_stores/integrations/) like so:
+
+```tsx
+const corpus = new LangchainCorpus(await getVectorStore());
+```
+
+Here is an example where we build a DocsQA component from AI.JSX documentation using Pinecone:
+
+```tsx
+async function getVectorStore(): Promise<VectorStore> {
+  const client = new PineconeClient();
+  await client.init({
+    apiKey: pineconeConfig.apiKey,
+    environment: pineconeConfig.environment,
+  });
+  const pineconeIndex = client.Index(pineconeConfig.index);
+
+  // For this example, we will index the ai-jsx docs.
+  const loader = new DirectoryLoader('../../packages/docs/docs/', {
+    '.md': (path) => new TextLoader(path),
```

Comment:
We should include the `import`s in the example.

Reply:
Good point. That's only for external imports, right?

Reply:
Basically, I want people to be able to paste examples into their editor and have it work. For pieces that are internal and common, like `ChatCompletion`, it's probably generally unnecessary. But yeah, for external imports, it seems good.